tests/5.1/runtime_calls/test_teams_region_routines.c

```diff
@@ -0,0 +1,88 @@
+//===------------ test-teams-region-routines.c ----------------------------===//
+//
+// OpenMP API Version 5.1 Aug 2021
+//
+// Section 3.4.3 from
+// https://www.openmp.org/wp-content/uploads/OpenMP-API-Specification-5-1.pdf
+//
+// Test for teams-region-routines added to 5.4. This routine sets number of
+// teams and threads to be used by default
+// in teams regions that do not specify them in the clause: 
+// omp_set_num_teams           and   omp_get_max_teams 
+// omp_set_teams_thread_limit  and   omp_get_teams_thread_limit
+// based on the 5.0/teams/test_teams.c test
+//
+//===----------------------------------------------------------------------===//
+
+#include <stdio.h>
+#include <omp.h>
+#include "ompvv.h"
+
+
+int test_teams_region_routines()
+{
+  int num_teams[OMPVV_NUM_TEAMS_DEVICE];
+  int num_threads[OMPVV_NUM_THREADS_DEVICE];
+  int errors[2] = {0,0};
+  int is_offloading;
+
+
+  for (int x = 0; x < OMPVV_NUM_TEAMS_DEVICE; ++x) {
+    num_teams[x] = -99;
+  }
+
+  for (int x = 0; x < OMPVV_NUM_THREADS_DEVICE; ++x) {
+    num_threads[x] = -99;
+  }
+
+  //sets runtime test
+  omp_set_num_teams(OMPVV_NUM_TEAMS_DEVICE);
+  omp_set_teams_thread_limit(OMPVV_NUM_THREADS_DEVICE);
+
+  //team error if the max teams doesn't get the set value
+  OMPVV_TEST_AND_SET_VERBOSE(errors[0], omp_get_max_teams() != OMPVV_NUM_TEAMS_DEVICE);
+
+  //thread error if the max teams doesn't get the set value
+  OMPVV_TEST_AND_SET_VERBOSE(errors[1], omp_get_teams_thread_limit() != OMPVV_NUM_THREADS_DEVICE);
+
+  // The test will also check if the correct team and thread values are passed
+#pragma omp teams 
+  {
+    num_teams[omp_get_team_num()] = omp_get_num_teams();
+#pragma omp parallel master
```

Comment:
change master to "masked".

Reply:
When I make this change, I don't get the test to pass, because
in line 53: 
num_threads[omp_get_team_num()]= omp_get_num_threads();
do not get the value of the number of threads
Is it ok if we write it as:
    #pragma omp parallel
    if (omp_get_thread_num() == 0)
      num_threads[omp_get_team_num()]= omp_get_num_threads();

Reply:
Which compiler are you testing with ? Maybe they do not have "masked" supported yet. Though in that case, I would think, there would be a compile time error. 

If while testing with master you are getting the correct answer, just replace with masked as it should be supported in Spec 5.1.

Reply:
> Which compiler are you testing with ? Maybe they do not have "masked" supported yet. Though in that case, I would think, there would be a compile time error.

Two remarks: For GCC, 'masked' is supported since GCC 12 (released this spring). And regarding unknown pragmas, C compilers tend to simply ignore them – and only warn for them either by default (like with Clang) or only if some warning flag is specified (such as -Wunknown-pragmas, implies by -Wall, in case of GCC). – Otherwise, the variant as commited compiles and shows "Test passed on the host." with GCC 13/mainline.